### PR TITLE
Utilize new tables on fingerprints and expired tokens view

### PR DESCRIPTION
**Pull request by NotMyFault**
The change proposed adapts the new table style on a couple more tables:

### Proposed changelog entries

* Use modern table design for "Recorder Fingerprints" and "Legacy API Tokens" tables.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/resources/jenkins/security/apitoken/LegacyApiTokenAdministrativeMonitor/manage.jelly
Comment (daniel-beck) [inline core/src/main/resources/jenkins/security/apitoken/LegacyApiTokenAdministrativeMonitor/manage.jelly:39]: I may have mentioned it before, but wouldn't it be more future proof to make these into Jelly tags we can rewrite as needed for future updates? This will be getting copied & pasted all over the place otherwise.

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
changelog entry?

**Comment by NotMyFault** [comment]
> changelog entry?

Added a one liner that sums it up.

File: core/src/main/resources/jenkins/security/apitoken/LegacyApiTokenAdministrativeMonitor/manage.jelly
Comment (timja) [reply core/src/main/resources/jenkins/security/apitoken/LegacyApiTokenAdministrativeMonitor/manage.jelly:39]: it could possibly be added now

something like?
```
l:table
l:tableHeader
l:tableBody
l:tableRow
l:tableFooter
```

Comment (fqueiruga) [reply core/src/main/resources/jenkins/security/apitoken/LegacyApiTokenAdministrativeMonitor/manage.jelly:39]: I think it would be great!

**Review by fqueiruga — APPROVED** [review]

File: core/src/main/resources/hudson/tasks/Fingerprinter/FingerprintAction/index.jelly
Comment (janfaracik) [inline core/src/main/resources/hudson/tasks/Fingerprinter/FingerprintAction/index.jelly:51]: Looks great to me :) Only thing I've spotted is this link is missing `class="jenkins-table__link"`

**Review by janfaracik — APPROVED** [review]

File: core/src/main/resources/jenkins/security/apitoken/LegacyApiTokenAdministrativeMonitor/manage.jelly
Comment (janfaracik) [reply core/src/main/resources/jenkins/security/apitoken/LegacyApiTokenAdministrativeMonitor/manage.jelly:39]: Looks good to me!

File: core/src/main/resources/hudson/tasks/Fingerprinter/FingerprintAction/index.jelly
Comment (NotMyFault) [reply core/src/main/resources/hudson/tasks/Fingerprinter/FingerprintAction/index.jelly]: Good catch!

File: core/src/main/resources/jenkins/security/apitoken/LegacyApiTokenAdministrativeMonitor/manage.jelly
Comment (timja) [reply core/src/main/resources/jenkins/security/apitoken/LegacyApiTokenAdministrativeMonitor/manage.jelly:39]: you up for trying that @NotMyFault? could be done in this PR as an example?

Comment (NotMyFault) [reply core/src/main/resources/jenkins/security/apitoken/LegacyApiTokenAdministrativeMonitor/manage.jelly:39]: Sounds like a great idea! Though unfortunately, I don't have as much spare time as I'd love to have to work on features like that, and I didn't work much with Jelly before, so I'd take me some time to get into it 😅

Comment (timja) [reply core/src/main/resources/jenkins/security/apitoken/LegacyApiTokenAdministrativeMonitor/manage.jelly:39]: i can prob push something in to your branch tomorrow or next week if that's okay?

Comment (NotMyFault) [reply core/src/main/resources/jenkins/security/apitoken/LegacyApiTokenAdministrativeMonitor/manage.jelly:39]: For sure, go ahead!

Comment (uhafner) [reply core/src/main/resources/jenkins/security/apitoken/LegacyApiTokenAdministrativeMonitor/manage.jelly:39]: Yes, using some tags would be very helpful. 

(I think it might even make sense to go another step deeper and replace the whole page with a template view that contains the header and the table and takes care of the  responsiveness. And the actual component just provides the data. I used that approach in the bootstrap and [data-tables-api-plugin](https://github.com/jenkinsci/data-tables-api-plugin#tables-with-dynamic-model-based-content) and my plugin views only provide the Java based object model, see [Warnings Issues table](https://github.com/jenkinsci/warnings-ng-plugin/blob/bfef1b523e09fbd654a219f0e058464570ca0101/plugin/src/main/resources/io/jenkins/plugins/analysis/core/model/IssuesDetail/filled.jelly#L111)). So when I hopefully find some time to apply the new tables styles to the bootstrap and data-tables plugins then all plugins that use them will get those changes without additional work.)

**Review by uhafner — APPROVED** [review]

**Comment by timja** [comment]
I do plan to get to this, couple of other things ahead of it, no need to hold this up.

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
